examples/access_traits/example_cuda_access_traits.cpp

```diff
@@ -65,21 +65,27 @@ int main(int argc, char *argv[])
 
     std::iota(std::begin(a), std::end(a), 1.0);
 
-    cudaMemcpy(d_a, a.data(), sizeof(a), cudaMemcpyHostToDevice);
+    cudaStream_t stream;
+    cudaStreamCreate(&stream);
+    cudaMemcpyAsync(d_a, a.data(), sizeof(a), cudaMemcpyHostToDevice, stream);
 
-    using device_type = Kokkos::Cuda::device_type;
-    ArborX::BVH<device_type> bvh{PointCloud{d_a, d_a, d_a, N}};
+    Kokkos::Cuda cuda{stream};
+    ArborX::BVH<Kokkos::CudaSpace> bvh{cuda, PointCloud{d_a, d_a, d_a, N}};
 
-    Kokkos::View<int *, device_type> indices("indices", 0);
-    Kokkos::View<int *, device_type> offset("offset", 0);
-    bvh.query(Spheres{d_a, d_a, d_a, d_a, N}, indices, offset);
+    using DeviceType = Kokkos::Cuda::device_type;
+    Kokkos::View<int *, DeviceType> indices("indices", 0);
+    Kokkos::View<int *, DeviceType> offset("offset", 0);
+    bvh.query(cuda, Spheres{d_a, d_a, d_a, d_a, N}, indices, offset);
 
-    Kokkos::parallel_for(N, KOKKOS_LAMBDA(int i) {
-      for (int j = offset(i); j < offset(i + 1); ++j)
-      {
-        printf("%i %i\n", i, indices(j));
-      }
-    });
+    Kokkos::parallel_for(Kokkos::RangePolicy<Kokkos::Cuda>(cuda, 0, N),
+                         KOKKOS_LAMBDA(int i) {
+                           for (int j = offset(i); j < offset(i + 1); ++j)
+                           {
+                             printf("%i %i\n", i, indices(j));
+                           }
+                         });
+
+    cudaStreamDestroy(&stream);
```

Comment:
```suggestion
    cudaStreamDestroy(stream);
```

Reply:
Does one need a fence after launching parallel_for but before destroying the stream?

Reply:
No. If the stream is doing work the function returns immediately and the ressource are released automatically when it is done.

Reply:
Per [CUDA documentation](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__STREAM.html#group__CUDART__STREAM_1gfda584f1788ca983cb21c5f4d2033a62):
```
In case the device is still doing work in the stream stream when cudaStreamDestroy() is called, the function will return immediately and the resources associated with stream will be released automatically once the device has completed all work in stream.
```